uaclient/mainwindow.py

```diff
@@ -205,6 +219,122 @@ def _update_subscription_model(self, node, value, timestamp):
+
+        # connect Apply button
+        self.window.ui.buttonApply.clicked.connect(self.restartTimer)
+        self.restartTimer()
+
+
+    def restartTimer(self):
+        # stop current timer, if it exists
+        if hasattr(self,'timer') and self.timer.isActive():
+            self.timer.stop()
+
+        # define the number of polls displayed in graph
+        self.N = self.window.ui.spinBoxNumberOfPoints.value()
+        self.ts = np.arange(self.N)
+        # define the poll intervall
+        self.intervall = self.window.ui.spinBoxIntervall.value()*1000
+
+        # overwrite current channel buffers with zeros of current length and add to curves again
+        for i,channel in enumerate(self._channels):
+            self._channels[i] = np.zeros(self.N)
+            self._curves[i].setData(self._channels[i])
+
+        # starting new timer
+        self.timer = QTimer()
+        self.timer.setInterval(self.intervall)
+        self.timer.timeout.connect(self.pushtoGraph)
+        self.timer.start()
+        
+    @trycatchslot
+    def _add_node_to_channel(self,node=None):
+        # TODO: check if node is of number type
+        if not isinstance(node, Node):
+            node = self.window.get_current_node()
+            if node is None:
+                return
+        if node not in self._node_list:
+            dtype = node.get_attribute(ua.AttributeIds.DataType)
+
+            dtypeStr = ua.ObjectIdNames[dtype.Value.Value.Identifier]
+
+            #TODO: arrays are detected by checking if the value is a list, but should use get_array_dimensions
+            #TODO:  instead. Unfortunately get_array_dimensions always returns an empty list.
+            if dtypeStr in self.acceptedDatatypes and not isinstance(node.get_value(),list):
+                self._node_list.append(node)
+                displayName = node.get_display_name().Text.decode('utf-8')
+                colorIndex = len(self._node_list) % len(self.colorCycle)
+                self._curves.append(self.pw.plot(pen=pg.mkPen(color=self.colorCycle[colorIndex],width=3,style=Qt.SolidLine), name=displayName))
+                # set initial data to zero
+                self._channels.append(np.zeros(self.N)) # init data sequence with zeros
+                # add the new channel data to the new curve
+                self._curves[-1].setData(self._channels[-1])
+                logger.info("Variable %s added to graph", displayName)
+
+            else:
+                logger.info("Variable cannot be added to graph because it is of type %s or an array", dtypeStr)
+
+
+    @trycatchslot
+    def _remove_node_from_channel(self,node=None):
+        if not isinstance(node, Node):
+            node = self.window.get_current_node()
+            if node is None:
+                return
+        if node in self._node_list:
+            idx = self._node_list.index(node)
+            self._node_list.pop(idx)
+            displayName = node.get_display_name().Text.decode('utf-8')
```

Comment:
Cant confirm this. Last fix on master of https://github.com/FreeOpcUa/python-opcua still seems to return byte string. Nevertheless, because this is to be changed, I decided to remove the decoding and wait for the fix!